sites/public/lib/appAutofill.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  Address,
+  Application,
+  ApplicationStatus,
+  ApplicationSubmissionType,
+  Language,
+} from "@bloom-housing/backend-core/types"
+
+class AutofillCleaner {
+  application: Application = null
+
+  constructor(application: Application) {
+    this.application = application
+  }
+
+  clean() {
+    // prettier-ignore
+    this.
+      addDefaults().
+      removeAdditionalKeys().
+      removeLiveWorkAddresses()
+
+    return this.application
+  }
+
+  addDefaults() {
+    ;["id", "createdAt", "updatedAt", "deletedAt", "listing", "submissionDate"].forEach((key) => {
+      delete this.application[key]
+    })
+
+    this.application["confirmationId"] = "" // only used on frontend
+    this.application["completedSections"] = 0 // only used on frontend
+    this.application["autofilled"] = true // only used on frontend
+    this.application.submissionType = ApplicationSubmissionType.electronical
+    this.application.language = Language.en
+    this.application.acceptedTerms = false
+    this.application.status = ApplicationStatus.submitted
+    this.application.preferences = []
+
+    return this
+  }
+
+  removeAdditionalKeys() {
+    const unsetIdentifiers = (obj: { id: string; createdAt: Date; updatedAt: Date }) => {
+      delete obj.id
+      delete obj.createdAt
+      delete obj.updatedAt
+    }
+
+    unsetIdentifiers(this.application.accessibility)
+    unsetIdentifiers(this.application.applicant)
+    unsetIdentifiers(this.application.mailingAddress)
+
+    if (this.application.alternateAddress) unsetIdentifiers(this.application.alternateAddress)
+
+    this.application.householdMembers
+      .sort((a, b) => a.orderId - b.orderId)
+      .forEach((member, index) => {
+        unsetIdentifiers(member)
+        member.orderId = index
+        if (member.address) unsetIdentifiers(member.address)
+        if (member.workAddress) unsetIdentifiers(member.workAddress)
+      })
+    unsetIdentifiers(this.application.demographics)
+
+    if (this.application.alternateContact) {
+      unsetIdentifiers(this.application.alternateContact)
+      if (this.application.alternateContact.mailingAddress) {
+        unsetIdentifiers(this.application.alternateContact.mailingAddress)
+      }
+    }
+
+    return this
+  }
+
+  removeLiveWorkAddresses() {
+    this.application.applicant.workInRegion = null
+    this.application.applicant.workAddress = {
+      street: "",
+      street2: "",
+      city: "",
+      state: "",
+      zipCode: "",
+      county: "",
+      latitude: null,
+      longitude: null,
```

Comment:
I think could be better to load defaults from the `blankApplication.ts`.

Reply:
Good point, I could load that in and copy the object over.